lib/iris/cube.py

```diff
@@ -3055,7 +2984,32 @@ def _order(array):
 
         return cube_xml_element
 
-    def copy(self, data=None):
+    def replace(self, data, dtype=None, fill_value=None):
+        """
+        Perform an in-place replacement of the cube data.
+
+        Args:
+
+        * data:
+            Replace the data of the cube with the provided data payload.
+
+        Kwargs:
+
+        * dtype:
+            Replacement for the intended dtype of the realised lazy data.
+
+        * fill_value:
+            Replacement for the cube data fill-value.
+
+        .. note::
+            Data replacement alone will clear the intended dtype
+            of the realised lazy data and the fill-value.
+
+        """
+        self._data_manager.replace(data, realised_dtype=dtype)
+        self.fill_value = fill_value
```

Comment:
It's used in several places for testing, namely:
* `tests/test_concatenate.py`
* `tests/test_interpolation.py`
* `tests/unit/analysis/regrid/test_RectilinearRegridder.py`

Yes, it's yet another method on the `Cube`, but it tightly couples the `fill_value` and `dtype` along with the `data`, supporting the action of replacing data directly (since the user simply can't achieve the same end with `cube.data = data` anymore, wrt managing the `realised_dtype` and `fill_value`) and as an alternative, it just might not be appropriate to copy the whole cube, but with different data. That could be quite expensive.

I think `Cube.replace` addresses a real use case and has tangible benefits, but please talk me down if you think it's inappropriate.